### PR TITLE
Rename telemetry prop

### PR DESCRIPTION
**Pull request by markfields**
The office telemetry logging code doesn't like dashes in telemetry prop names.  I thought I fixed this in #8449 but the fix was incomplete.  That PR addressed the error case, but I didn't realize that for successful requests we were stashing the same telemetry prop on the internal response object and later logging it as well.

So I fixed that, and renamed the property bag to be more clear that it's for logging and not for business logic.

### CODE REVIEW

**Comment by markfields** [comment]
@abrahamrq FYI

**Comment by vladsud** [comment]
Feels like the "fix" should be on FFX side. I.e. nothing in the system prevents some layer using dashes, and that maybe fine for other systems.

File: packages/drivers/odsp-driver/src/createFile.ts [hunk 1]
```diff
@@ -161,7 +161,7 @@ export async function createNewEmptyFluidFile(
                 }
                 event.end({
                     headers: Object.keys(headers).length !== 0 ? true : undefined,
-                    ...fetchResponse.commonSpoHeaders,
+                    ...fetchResponse.telemetryProps,
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/createFile.ts:164]: I'd personally prefer an old name. "Telemetry props" is the type, it feels like Hungarian notation. "Spo headers" is what they represent.
If you want to rename, then likely you want to articulate the purpose, i.e., "fetchTelemetryPropsToLog" or something like that.

**Review by vladsud — APPROVED** [review]

File: packages/drivers/odsp-driver/src/createFile.ts (hunk 1 above)
Comment (markfields) [reply packages/drivers/odsp-driver/src/createFile.ts:164]: People are putting other stuff not headers on there

**Pull request merged**